### ISSUE TITLE
[a11y]: [TreeView Component] A11y (Accessibility) violation

### ISSUE THREAD

**Issue by loftywaif002** [carbon-design-system/carbon#19099]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.76.0

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Accessibility Access checker

### Description

Hi Carbon team,

As part of an IBM-wide requirement, a requirement is for DataStage UI (DataStage/ds-canvas) to be accessibility compliant, via the [IBM Equal Access Accessibility Checker tool](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp?hl=en-US).

### Reproduction/example

https://stackblitz.com/edit/github-i4v7cuer

### Steps to reproduce

1. open the stackblitz
2. `run npm install `
3. `npx webpack serve`
4. Click on the top right icon to "`Open Preview in a new tab`"
5. Run the accessibility checker on the page, you should see something like below :

![Image](https://github.com/user-attachments/assets/290b9bf0-8121-4bae-91af-b97b30065520)

Video: 

https://www.dropbox.com/scl/fi/cnn8wlntd5zeur2fpp0tl/TreeView-Accessibility.mov?rlkey=7wwil4c58t1ljgxwppc5y6okv&st=d7ao9m4o&dl=0

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
